From 04ab2565b6918637c7c6c61cca771a5156730507 Mon Sep 17 00:00:00 2001
From: xubing <bing.xu@intel.com>
Date: Thu, 14 Mar 2024 15:47:26 +0800
Subject: [PATCH] Fix JavaCarsh on com.android.gallery3d

User click the setting menu, App should do nothing
when there is no primary image.

Test: Run monkey test and observe, no crash happen.

Tracked-On: OAM-116823
Signed-off-by: xubing <bing.xu@intel.com>
---
 .../gallery3d/filtershow/FilterShowActivity.java     | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/com/android/gallery3d/filtershow/FilterShowActivity.java b/src/com/android/gallery3d/filtershow/FilterShowActivity.java
index 0d25186ba..418be9160 100644
--- a/src/com/android/gallery3d/filtershow/FilterShowActivity.java
+++ b/src/com/android/gallery3d/filtershow/FilterShowActivity.java
@@ -1044,6 +1044,10 @@ public class FilterShowActivity extends FragmentActivity implements OnItemClickL
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.undoButton: {
+                //we should do nothing if there is no primary image
+                if (mPrimaryImage == null) {
+                    return false;
+                }
                 HistoryManager adapter = mPrimaryImage.getHistory();
                 int position = adapter.undo();
                 mPrimaryImage.onHistoryItemClick(position);
@@ -1052,6 +1056,10 @@ public class FilterShowActivity extends FragmentActivity implements OnItemClickL
                 return true;
             }
             case R.id.redoButton: {
+                //we should do nothing if there is no primary image
+                if (mPrimaryImage == null) {
+                    return false;
+                }
                 HistoryManager adapter = mPrimaryImage.getHistory();
                 int position = adapter.redo();
                 mPrimaryImage.onHistoryItemClick(position);
@@ -1299,6 +1307,10 @@ public class FilterShowActivity extends FragmentActivity implements OnItemClickL
     }
 
     void resetHistory() {
+        //we should do nothing if there is no primary image
+        if (mPrimaryImage == null) {
+            return;
+        }
         HistoryManager adapter = mPrimaryImage.getHistory();
         adapter.reset();
         HistoryItem historyItem = adapter.getItem(0);
-- 
2.34.1

